sample-compose/app/src/main/java/co/nimblehq/sample/compose/extensions/FlowExt.kt

```diff
@@ -0,0 +1,19 @@
+package co.nimblehq.sample.compose.extensions
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import kotlinx.coroutines.flow.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+@SuppressLint("ComposableNaming")
+@Composable
+fun <T> Flow<T>.collectAsEffect(
+    context: CoroutineContext = EmptyCoroutineContext,
+    block: (T) -> Unit,
+) {
+    LaunchedEffect(key1 = Unit) {
+        onEach(block).flowOn(context).launchIn(this)
+    }
+}
```

Comment:
As the team raised about [allowing modification for the LaunchedEffect's key1](https://github.com/nimblehq/android-templates/discussions/408#discussioncomment-5060034), I also think we should support passing value for this key as it can inform `LaunchedEffect to whether the coroutine should restart or not when the key1 parameter value changes. 🙏

You can refer to the following link as a good example use case regarding the need to modify this key e.g., re-executing the API:
https://betterprogramming.pub/jetpack-compose-side-effects-launchedeffect-with-example-99c2f51ff463

Reply:
@minhnimble Since we used `onEach(block).flowOn(context).launchIn(this)`, we don't need to expose `key1` for value changes to re-subscribing. The `onEach(block).flowOn(context).launchIn(this)` is automatically emit new value from Flow itself without the help from `LaunchedEffect(key1 = )` 😄 

I already explained the unnecessary of exposing `key1` in PR's description.
![image](https://user-images.githubusercontent.com/16315358/228749347-b5c3722a-468c-44ea-9ba3-eedc88ecd635.png)

Reply:
Okay, I got it now. Previously I had a look through the PR description, but it was still ambiguous for me that you just mentioned:

> We don't need to expose a key to replace the Unit, because we already used `onEach(block).flowOn(context).launchIn(this)`.

=> Maybe you should explain a bit deeper in the description into how the fact that `onEach(block).flowOn(context).launchIn(this)` can automatically emit new values from Flow itself without the help from `LaunchedEffect` with key. 🙏

Reply:
@minhnimble Yeah. Updated 👍